src/app/dataset-view/dataset.permissions.service.ts

```diff
@@ -11,10 +11,20 @@ import { DatasetPermissionsFragment } from "../api/kamu.graphql.interface";
 @Injectable({ providedIn: "root" })
 export class DatasetPermissionsService {
     public shouldAllowSettingsTab(datasetPermissions: DatasetPermissionsFragment): boolean {
-        return datasetPermissions.permissions.canDelete || datasetPermissions.permissions.canRename;
+        return (
+            datasetPermissions.permissions.envVars.canView ||
+            datasetPermissions.permissions.collaboration.canView ||
+            datasetPermissions.permissions.general.canDelete ||
+            datasetPermissions.permissions.general.canRename ||
+            datasetPermissions.permissions.general.canSetVisibility
+        );
```

Comment:
So, we will have to modify and modify this each time we add some new tab.
This is too error-prone, let's discuss how we can improve this.

Reply:
Discussed this solution